Isodose/Logic/vtkSlicerIsodoseModuleLogic.cxx

```diff
@@ -880,10 +799,57 @@ void vtkSlicerIsodoseModuleLogic::CreateIsodoseSurfaces(vtkMRMLIsodoseNode* para
     this->InvokeEvent(vtkSlicerRtCommon::ProgressUpdated, (void*)&progress);
   } // For all isodose levels
 
-  // Update dose color table based on isodose
-  this->UpdateDoseColorTableFromIsodose(parameterNode);
+  // update appended isosurfaces
+  append->Update();
+
+  vtkPolyData* isoSurfaces = append->GetOutput();
+  if (isoSurfaces)
+  {
+    isoSurfaces->GetPointData()->SetScalars(colors);
+
+    vtkMRMLModelDisplayNode* displayNode = vtkMRMLModelDisplayNode::SafeDownCast(
+      scene->AddNewNodeByClass("vtkMRMLModelDisplayNode"));
+    displayNode->Visibility2DOn();
+    displayNode->VisibilityOn();
+    displayNode->SetActiveScalarName("isolevels");
+    displayNode->SetAutoScalarRange(true);
+    displayNode->SetAndObserveColorNodeID(colorTableNode->GetID());
+    displayNode->SetScalarVisibility(true);
+
+    // Disable backface culling to make the back side of the model visible as well
+    displayNode->SetBackfaceCulling(0);
+
+    vtkNew<vtkMRMLModelNode> isodoseModelNode;
+    std::string uniqueName = scene->GenerateUniqueName(vtkSlicerIsodoseModuleLogic::ISODOSE_MODEL_NODE_NAME_PREFIX);

```

Comment:
This is where you create the ambiguous name. Please use the dose volume's name as prefix and just add a postfix saying this is the isodose levels model.